plugin/plugin.go

```diff
@@ -0,0 +1,288 @@
+	}
+
+	if len(files) == 0 {
+		if args.PluginFailIfNoResults {
+			return errors.New("no TestNG XML report files found. Check the report file pattern")
+		}
+		logrus.Warn("No TestNG XML report files found, continuing execution as PluginFailIfNoResults is false")
+		return nil
+	}
+
+	var aggregatedResults Results
+
+	for _, file := range files {
+		results, err := processFile(file)
+		if err != nil {
+			logger := logrus.WithField("File", file).WithError(err)
+			logger.Error("Error processing file")
+			return errors.New("failed to process file: " + err.Error())
+		}
+		aggregatedResults.Total += results.Total
+		aggregatedResults.Failures += results.Failures
+		aggregatedResults.Skipped += results.Skipped
+		aggregatedResults.DurationMS += results.DurationMS
+	}
+
+	// Log aggregated results
+	logrus.Infof("\n===============================================")
+	logrus.Infof("\nTotal Tests Results: %d | Failures: %d | Skips: %d | Duration: %.2f ms", aggregatedResults.Total, aggregatedResults.Failures, aggregatedResults.Skipped, aggregatedResults.DurationMS)
+	logrus.Infof("\n===============================================")
+
+	// Validate thresholds at the aggregate level
+	if err := validateThresholds(aggregatedResults, args); err != nil {
+		logger := logrus.WithFields(logrus.Fields{
+			"Total Tests": aggregatedResults.Total,
+			"Failures":    aggregatedResults.Failures,
+			"Skipped":     aggregatedResults.Skipped,
+			"DurationMS":  aggregatedResults.DurationMS,
+		})
+		logger.Error(err.Error())
+		return err
+	}
+
+	return nil
+}
+
+// locateFiles identifies files matching the given pattern.
+func locateFiles(pattern string) ([]string, error) {
+	matches, err := filepath.Glob(pattern)
+	if err != nil {
+		logger := logrus.WithError(err).WithField("Pattern", pattern)
+		logger.Error("Error occurred while searching for files")
+		return nil, errors.New("failed to search for files: " + err.Error())
+	}
+	if len(matches) == 0 {
+		return nil, errors.New("no files found matching the report filename pattern")
+	}
+	return matches, nil
+}
+
+// processFile reads a TestNG XML report and logs details.
+func processFile(filename string) (Results, error) {
+	logrus.Infof("Processing file: %s", filename)
+
+	data, err := os.ReadFile(filename)
+	if err != nil {
+		logger := logrus.WithError(err).WithField("File", filename)
+		logger.Error("Failed to read file")
+		return Results{}, errors.New("failed to read file: " + err.Error())
+	}
+
+	var report TestNGReport
+	if err := xml.Unmarshal(data, &report); err != nil {
+		logger := logrus.WithError(err).WithField("File", filename)
+		logger.Error("Failed to parse TestNG XML")
+		return Results{}, errors.New("failed to parse TestNG XML: " + err.Error())
+	}
+
+	return logTestNGReportDetails(report), nil
+}
+
+// logTestNGReportDetails logs the details of a TestNG report and returns the aggregated results.
+func logTestNGReportDetails(report TestNGReport) Results {
```

Comment:
add check to handle invalid durations by adding a warning log for invalid or missing DurationMS values and also log suite-level summaries 

again create helper functions to reduce duplication in aggregation logic

once the above completed, add uts for aggregation logic